src/platform/plugins/private/presentation_panel/public/panel_component/panel_header/use_hover_actions_styles.tsx

```diff
@@ -0,0 +1,142 @@
+        transition: ${euiTheme.animation.extraFast} opacity ease-in,
+          ${euiTheme.animation.extraFast} z-index linear,
+          ${euiTheme.animation.extraFast} visibility linear;
+        transition-delay: ${euiTheme.animation.fast};
+      }
+
+      &:hover .embPanel__hoverActions,
+      &:has(:focus-visible) .embPanel__hoverActions,
+      &.embPanel__hoverActionsAnchor--lockHoverActions .embPanel__hoverActions,
+      .embPanel__hoverActions:hover,
+      .embPanel__hoverActions:has(:focus-visible) {
+        z-index: ${euiTheme.levels.menu};
+        opacity: 1;
+        visibility: visible;
+        transition: none; // apply transition delay on hover out only
+      }
+    `;
+  }, [euiTheme, showBorder, isEditMode]);
+
+  const hoverActionStyles = useMemo(() => {
+    const singleWrapperStyles = css`
+      width: fit-content;
+      top: -${euiTheme.size.l} !important;
+      right: ${euiTheme.size.xs};
+      padding: var(--paddingAroundAction);
+
+      border-radius: ${euiTheme.border.radius.medium};
+      border: var(--internalBorderStyle);
+      background-color: ${euiTheme.colors.backgroundBasePlain};
+      grid-template-columns: max-content;
+
+      & > * {
+        // undo certain styles on all children so that parent takes precedence
+        border: none !important;
+        padding: 0px !important;
+        border-radius: unset !important;
+        background-color: transparent !important;
+        height: unset !important;
+      }
+    `;
+
+    return css`
+      --paddingAroundAction: calc(${euiTheme.size.xs} - 1px);
+
+      pointer-events: none; // prevent hover actions wrapper from blocking interactions with other panels
+
+      width: 100%;
+      height: ${euiTheme.size.xl};
+      display: flex;
+      align-items: center;
+      justify-content: space-between;
+      padding: 0px ${euiTheme.size.m};
+
+      & > * {
+        // apply styles to all children
+        display: flex;
+        height: ${euiTheme.size.xl};
+        flex: 0; // do not grow
+        pointer-events: all; // re-enable pointer events for non-breakpoint children
+        background-color: ${euiTheme.colors.backgroundBasePlain};
+        border: var(--internalBorderStyle);
+        border-bottom: 0px;
+        padding: var(--paddingAroundAction);
+        padding-bottom: 0px;
+        border-top-left-radius: ${euiTheme.border.radius.medium};
+        border-top-right-radius: ${euiTheme.border.radius.medium};
+      }
+
+      // shrink down to single wrapped element with no breakpoint when panel gets small
+      @container hoverActionsAnchor (width < 200px) {
+        ${singleWrapperStyles}
+      }
+
+      // when Dashboard is in fullscreen mode, combine all floating actions on first row and nudge them down;
+      // if the panel is **not** on the first row but it is expanded in fullscreen mode, do the same thing
+      .dshDashboardViewportWrapper--isFullscreen .dshDashboardGrid__item[data-grid-row='0'] &,
```

Comment:
This isn't necessarily something to change, but this code uses the nudged actions in full screen even when there _are_ controls.

<img width="771" alt="Screenshot 2025-02-04 at 12 50 24 PM" src="https://github.com/user-attachments/assets/ee938c25-3842-4efc-ac33-5ae4f56cfc71" />

Reply:
Yup, I left it this way intentionally to try to keep as much dashboard-specific logic out of the hover actions as possible. I could easily re-add this check, because I kind of went back and forth on whether this extra complexity was worth it for this edge case and I'm not really sold on either direction haha :) cc @andreadelrio

Reply:
It's such a small detail that I would lean towards merging as is.